dispatcher/publish.py

```diff
@@ -1,19 +1,30 @@
-import inspect
-import json
 import logging
-import time
-from uuid import uuid4
+from typing import Optional
 
-from dispatcher.utils import DuplicateBehavior
+from dispatcher.registry import DispatcherMethodRegistry
+from dispatcher.registry import registry as default_registry
+from dispatcher.utils import DispatcherCallable
 
 logger = logging.getLogger('awx.main.dispatch')
 
 
-def serialize_task(f):
-    return '.'.join([f.__module__, f.__name__])
+def _task_decorator(fn: DispatcherCallable, registry=default_registry, **kwargs) -> DispatcherCallable:
```

Comment:
should we allow or disallow explicitly positional arguments in this function?

Reply:
We _must_ allow `fn` as a positional argument because it's a decorator. I tried to add `, *` but something about that wasn't working. Now I'm not remembering what.

Reply:
Sure `fn`, I did not express myself well. I meant each other pos args (*args) or not supporting the `registry` as positional and/or enforce kwargs after `fn`

Reply:
Yes, 100%. We have never used positional kwargs to `@task` other than native decorator functionality. Anything other than the function should be strictly key word only. It seems that adding `*, ` is working for me now, so I'll push a commit with that.

And now that I look at this, I've also convinced myself that only `@task()` should be allowed, not `@task`.